torch/csrc/distributed/autograd/context/dist_autograd_context.cpp

```diff
@@ -48,6 +48,68 @@ DistAutogradContext::recvFunctions() const {
   return recvAutogradFunctions_;
 }
 
+void DistAutogradContext::accumulateGrad(
+    torch::autograd::Variable variable,
+    torch::Tensor grad) {
+  TORCH_INTERNAL_ASSERT(grad.defined());
+  TORCH_INTERNAL_ASSERT(variable.requires_grad());
+
+  std::lock_guard<std::mutex> guard(lock_);
+  auto it = accumulatedGrads_.find(variable);
+  if (it != accumulatedGrads_.end()) {
+    // Accumulate multiple grads on the same variable.
+    it->setValue(it->value() + grad);
```

Comment:
Any reason not to do it inplace?

Reply:
(Hmm, I guess when we insert the grad will alias with the input. Not sure if this causes trouble.)

Reply:
How would you do this in place? It looks like `DictEntryRef` only exposes a `setValue` method to modify the value.

Reply:
`it->value().add_(grad)`